erdgo/blockchain/proxy.go

```diff
@@ -25,15 +25,25 @@ const (
 	getNetworkStatusEndpoint     = "network/status/%v"
 )
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 // elrondProxy implements basic functions for interacting with an Elrond Proxy
 type elrondProxy struct {
 	proxyURL string
+	client   HTTPClient
 }
 
 // NewElrondProxy initializes and returns an ElrondProxy object
-func NewElrondProxy(url string) *elrondProxy {
+func NewElrondProxy(url string, client HTTPClient) *elrondProxy {
+	if client == nil {
```

Comment:
instead of this check we use the check.IfNill(client) call provided that he HTTPClient implements IsInterfaceNil.